src/main/java/com/cqse/teamscaleupload/TeamscaleUpload.java

```diff
@@ -71,9 +81,14 @@ public void validate(ArgumentParser parser) throws ArgumentParserException {
                         " upload data to Teamscale", parser);
             }
 
-            if (files == null && inputFile == null) {
-                throw new ArgumentParserException("You must either specify the paths of the coverage files as plain " +
-                        "arguments or provide them in an input file, see help for more information", parser);
+            if (files.isEmpty() && input == null) {
+                throw new ArgumentParserException("No report files provided." +
+                        " You must either specify the paths of the coverage files as command line" +
+                        " arguments or provide them in an input file via --input", parser);
+            }
+
+            if (format == null) {
+                throw new ArgumentParserException("Please provide the default report format with --format", parser);
```

Comment:
I made the format argument mandatory. It may not be necessary when all patterns have a corresponding [format] in the input file. However, it is much simpler if we do not have to check later if the default format is set.

Another option would be to automatically set a fallback format if the user does not specify one. This might, however, cause unexpected behavior when the user forgets to specify the format and files are then uploaded with the wrong fallback format. 

What do you think?

Reply:
That works for me, but instead of checking yourself manually please use the argument parser library to do the required check like here: https://github.com/cqse/teamscale-upload/blob/e97ab41ed16c16c9a47c3e47b3155a80bfc828c8/src/main/java/com/cqse/teamscaleupload/TeamscaleUpload.java#L125
`.required(true)`

This way it's consistent with the other parameters and highlighted as required in `--help`

Reply:
Actually, while reading the code I had another idea: how about we make it mandatory to specify a format in the file and only apply the `--format` to stuff given on the command line?
That would make the code simpler I think and we can clearly document this in `--help`. What do you think?